Required or recommended to have cosign.pub?

When signing something, you mention that it is recommended to have a cosign.pub. This is fine, but I don't have one in the repository which I'm testing. 
I try to set mine using the environment variable ``COSIGN_PUBLIC_KEY``.

**Problem:**
```Finished getting docker digest and tags
============================================================================================
Signing image
Sign image
Pushing signature to: ghcr.io/something
Verify signing
Error: loading public key: open cosign.pub: no such file or directory
main.go:46: error during command execution: loading public key: open cosign.pub: no such file or director
```
It does not seem to find a cosign.pub even though I added the environment variable.

**Assumption:**
I expect the software to check if the environment variable is set, if it is, it should be used, if not, it will try to look for a cosign.pub in the directory. If neither are set or found, it should give me the above error.

--- a/container_digest.sh
+++ b/container_digest.sh
@@ -78,10 +78,6 @@ then
 
   echo "Verify signing"
   cosign verify --key "$COSIGN_PUB" "$docker_registry_prefix"/"$imagename"@"${containerdigest}" 
-
-  echo "Cleanup"
-  rm "$COSIGN_KEY"
-  rm "$COSIGN_PUB"
 fi
 
 if [ -n "${SLSA_PROVENANCE}" ]
@@ -115,13 +111,6 @@ then
   then
     echo "Attaching SLSA Provenance with Cosign"
 
-    COSIGN_KEY=$(mktemp /tmp/cosign.XXXXXXXXXX) || exit 1
-    COSIGN_PUB=$(mktemp /tmp/cosign.XXXXXXXXXX) || exit 1
-
-    # COSGIN_PASSWORD should be passed as environment variable
-    echo "${COSIGN_PRIVATE_KEY}" > "$COSIGN_KEY"
-    echo "${COSIGN_PUBLIC_KEY}" > "$COSIGN_PUB"
-
     echo "Get predicate"
     jq .predicate < provenance.json > provenance-predicate.json
 
@@ -131,9 +120,6 @@ then
     echo "Verify predicate"
     cosign verify-attestation --key "$COSIGN_PUB" "$docker_registry_prefix"/"$imagename"@"${containerdigest}"
 
-    echo "Cleanup"
-    rm "$COSIGN_KEY"
-    rm "$COSIGN_PUB"
   fi
 fi
 
@@ -153,21 +139,17 @@ then
   then
     echo "Attaching SBOM  with Cosign"
 
-    COSIGN_KEY=$(mktemp /tmp/cosign.XXXXXXXXXX) || exit 1
-    COSIGN_PUB=$(mktemp /tmp/cosign.XXXXXXXXXX) || exit 1
-
-    # COSGIN_PASSWORD should be passed as environment variable
-    echo "${COSIGN_PRIVATE_KEY}" > "$COSIGN_KEY"
-    echo "${COSIGN_PUBLIC_KEY}" > "$COSIGN_PUB"
-
     echo "Attest SBOM"
     cosign attest --predicate sbom-spdx.json --type spdx --key "$COSIGN_KEY" "$docker_registry_prefix"/"$imagename"@"${containerdigest}"
 
     echo "Verify SBOM"
     cosign verify-attestation --key "$COSIGN_PUB" "$docker_registry_prefix"/"$imagename"@"${containerdigest}"
-
-    echo "Cleanup"
-    rm "$COSIGN_KEY"
-    rm "$COSIGN_PUB"
   fi
+fi
+
+if [ -n "${SIGN}" ]
+  then
+  echo "Cleanup"
+  rm "$COSIGN_KEY"
+  rm "$COSIGN_PUB"
 fi

--- a/container_digest.sh
+++ b/container_digest.sh
@@ -66,18 +66,22 @@ if [ -n "${SIGN}" ]
 then
   echo "Signing image"
 
+  COSIGN_KEY=$(mktemp /tmp/cosign.XXXXXXXXXX) || exit 1
+  COSIGN_PUB=$(mktemp /tmp/cosign.XXXXXXXXXX) || exit 1
+
   # COSGIN_PASSWORD should be passed as environment variable
-  echo "${COSIGN_PRIVATE_KEY}" > cosign.key
+  echo "${COSIGN_PRIVATE_KEY}" > "$COSIGN_KEY"
+  echo "${COSIGN_PUBLIC_KEY}" > "$COSIGN_PUB"
 
   echo "Sign image"
-  cosign sign --key cosign.key "$docker_registry_prefix"/"$imagename"@"${containerdigest}"
+  cosign sign --key "$COSIGN_KEY" "$docker_registry_prefix"/"$imagename"@"${containerdigest}"
 
   echo "Verify signing"
-  cosign verify --key cosign.pub "$docker_registry_prefix"/"$imagename"@"${containerdigest}" 
+  cosign verify --key "$COSIGN_PUB" "$docker_registry_prefix"/"$imagename"@"${containerdigest}" 
 
   echo "Cleanup"
-  rm cosign.key
-  rm cosign.pub
+  rm "$COSIGN_KEY"
+  rm "$COSIGN_PUB"
 fi
 
 if [ -n "${SLSA_PROVENANCE}" ]
@@ -110,22 +114,26 @@ then
   if [ -n "${SIGN}" ]
   then
     echo "Attaching SLSA Provenance with Cosign"
+
+    COSIGN_KEY=$(mktemp /tmp/cosign.XXXXXXXXXX) || exit 1
+    COSIGN_PUB=$(mktemp /tmp/cosign.XXXXXXXXXX) || exit 1
+
+    # COSGIN_PASSWORD should be passed as environment variable
+    echo "${COSIGN_PRIVATE_KEY}" > "$COSIGN_KEY"
+    echo "${COSIGN_PUBLIC_KEY}" > "$COSIGN_PUB"
+
     echo "Get predicate"
-    echo "${COSIGN_PRIVATE_KEY}" > cosign.key
     jq .predicate < provenance.json > provenance-predicate.json
 
     echo "Attest predicate"
-    # COSGIN_PASSWORD should be passed as environment variable
-    echo "${COSIGN_PRIVATE_KEY}" > cosign.key
-    cosign attest --predicate provenance-predicate.json --key cosign.key --type slsaprovenance "$docker_registry_prefix"/"$imagename"@"${containerdigest}"
+    cosign attest --predicate provenance-predicate.json --key "$COSIGN_KEY" --type slsaprovenance "$docker_registry_prefix"/"$imagename"@"${containerdigest}"
 
     echo "Verify predicate"
-    echo "${COSIGN_PUBLIC_KEY}" > cosign.pub
-    cosign verify-attestation --key cosign.pub "$docker_registry_prefix"/"$imagename"@"${containerdigest}"
+    cosign verify-attestation --key "$COSIGN_PUB" "$docker_registry_prefix"/"$imagename"@"${containerdigest}"
 
     echo "Cleanup"
-    rm cosign.key
-    rm cosign.pub
+    rm "$COSIGN_KEY"
+    rm "$COSIGN_PUB"
   fi
 fi
 
@@ -144,21 +152,22 @@ then
   if [ -n "${SIGN}" ]
   then
     echo "Attaching SBOM  with Cosign"
-    echo "${COSIGN_PRIVATE_KEY}" > cosign.key
+
+    COSIGN_KEY=$(mktemp /tmp/cosign.XXXXXXXXXX) || exit 1
+    COSIGN_PUB=$(mktemp /tmp/cosign.XXXXXXXXXX) || exit 1
+
+    # COSGIN_PASSWORD should be passed as environment variable
+    echo "${COSIGN_PRIVATE_KEY}" > "$COSIGN_KEY"
+    echo "${COSIGN_PUBLIC_KEY}" > "$COSIGN_PUB"
 
     echo "Attest SBOM"
-    # COSGIN_PASSWORD should be passed as environment variable
-    echo "${COSIGN_PRIVATE_KEY}" > cosign.key
-    cosign attest --predicate sbom-spdx.json --type spdx --key cosign.key "$docker_registry_prefix"/"$imagename"@"${containerdigest}"
+    cosign attest --predicate sbom-spdx.json --type spdx --key "$COSIGN_KEY" "$docker_registry_prefix"/"$imagename"@"${containerdigest}"
 
     echo "Verify SBOM"
-    echo "${COSIGN_PUBLIC_KEY}" > cosign.pub
-    cosign verify-attestation --key cosign.pub "$docker_registry_prefix"/"$imagename"@"${containerdigest}"
+    cosign verify-attestation --key "$COSIGN_PUB" "$docker_registry_prefix"/"$imagename"@"${containerdigest}"
 
     echo "Cleanup"
-    rm cosign.key
-    rm cosign.pub
+    rm "$COSIGN_KEY"
+    rm "$COSIGN_PUB"
   fi
 fi
-
-

